### DISCUSSION TITLE
Getting HyperHDR working with ws281x led strip

### DISCUSSION THREAD

**Discussion by RamoZack** [awawa-dev/HyperHDR#780] [Q&A]
tldr; 
I have tried a lot. LED lights will not come on at all. it is connected to my raspberry pi through the data wire connecting to GPIO 26.

2024-02-24T20:42:50.753Z [LEDDEVICE_WS281X] Device disabled, device 'ws281x' signals error: 'Failed to open. Error message: Hardware revision is not supported'

error message using raspberry pi 3B+

running this command will give me this revision: 

cat /proc/cpuinfo | grep Revision
Revision        : a020d4

![screenshotconfig](https://github.com/awawa-dev/HyperHDR/assets/54585072/e15a1783-7e3c-48b9-9e05-00dd61fa9b2b)

2024-02-24T20:42:50.158Z [MAIN] Database path: '/.hyperhdr/db/hyperhdr.db', readonlyMode = disabled
2024-02-24T20:42:50.158Z [MAIN] Starting HyperHdr - 19.0.0.0, (HEAD detached at b8c7f04) (Awawa-3141e80/b8c7f04-1676222403), built: Feb 12 2023:18:04:00
2024-02-24T20:42:50.158Z [MAIN] Set user data path to '/.hyperhdr'
2024-02-24T20:42:50.206Z [DB] Database opened: /.hyperhdr/db/hyperhdr.db
2024-02-24T20:42:50.413Z [SETTINGSMGR] Settings database initialized
2024-02-24T20:42:50.420Z [HYPERHDR] Sound device is disabled
2024-02-24T20:42:50.477Z [V4L2:AUTO] Cropping image: width=0 height=0; crop: left=0 right=0 top=0 bottom=0 
2024-02-24T20:42:50.478Z [V4L2:AUTO] Found capture device: /dev/video14
2024-02-24T20:42:50.479Z [V4L2:AUTO] Found capture device: /dev/video22
2024-02-24T20:42:50.481Z [V4L2:AUTO] Found capture device: /dev/video15
2024-02-24T20:42:50.482Z [V4L2:AUTO] Found capture device: /dev/video21
2024-02-24T20:42:50.483Z [V4L2:AUTO] Restarting video grabber is now blocked due to reloading of the configuration
2024-02-24T20:42:50.483Z [V4L2:AUTO] Cropping image: width=0 height=0; crop: left=0 right=0 top=0 bottom=0 
2024-02-24T20:42:50.484Z [V4L2:auto] (GrabberWrapper.cpp:75) CEC keycode. Start: 0, stop: 0
2024-02-24T20:42:50.484Z [V4L2:AUTO] (Grabber.cpp:259) setBrightnessContrastSaturationHue nothing changed
2024-02-24T20:42:50.484Z [V4L2:AUTO] (V4L2Grabber.cpp:125) setHdrToneMappingMode to: Disabled
2024-02-24T20:42:50.485Z [V4L2:AUTO] (Grabber.cpp:168) setFpsSoftwareDecimation to: 1
2024-02-24T20:42:50.485Z [SIGNAL_OLD] (DetectionManual.cpp:85) Signal detection area set to: 0.250000,0.250000 x 0.750000,0.750000
2024-02-24T20:42:50.485Z [SIGNAL_OLD] (DetectionManual.cpp:75) Signal threshold set to: {12, 12, 12} and frames: 200
2024-02-24T20:42:50.485Z [SIGNAL_AUTO] (DetectionAutomatic.cpp:382) Automatic signal detection -> errorTolerance: 9, modelTolerance: 90, sleepTime: 5000, wakeTime: 1000
2024-02-24T20:42:50.486Z [V4L2:AUTO] (Grabber.cpp:367) setDeviceVideoStandard preparing to restart video grabber. Old: '' new: 'auto'
2024-02-24T20:42:50.486Z [V4L2:AUTO] Delayed restart of the grabber due to change of selected device
2024-02-24T20:42:50.486Z [V4L2:AUTO] (Grabber.cpp:186) Force encoding to: NO_CHANGE (old: NO_CHANGE)
2024-02-24T20:42:50.486Z [V4L2:AUTO] setQFrameDecimation is now: disabled
2024-02-24T20:42:50.487Z [V4L2:auto] (GrabberWrapper.cpp:545) Frame cache is: enabled
2024-02-24T20:42:50.489Z [PIPEWIRE_SYSTEM:AUTO] Cropping image: width=0 height=0; crop: left=0 right=0 top=0 bottom=0 
2024-02-24T20:42:50.500Z [PIPEWIRE_SYSTEM:AUTO] Could not load Pipewire proxy library. Error: libpipewire-0.3.so.0: cannot open shared object file: No such file or directory
2024-02-24T20:42:50.501Z [DAEMON] The system doesn't support the Pipewire/Portal grabber
2024-02-24T20:42:50.501Z [X11_SYSTEM:AUTO] Cropping image: width=0 height=0; crop: left=0 right=0 top=0 bottom=0 
2024-02-24T20:42:50.542Z [X11_SYSTEM:AUTO] Loaded X11 proxy library for screen capturing
2024-02-24T20:42:50.553Z [DAEMON] The system doesn't support the X11 grabber
2024-02-24T20:42:50.554Z [FRAMEBUFFER_SYSTEM:AUTO] Cropping image: width=0 height=0; crop: left=0 right=0 top=0 bottom=0 
2024-02-24T20:42:50.555Z [DAEMON] The system doesn't support the FrameBuffer grabber
2024-02-24T20:42:50.686Z [SETTINGSMGR] Settings database initialized
2024-02-24T20:42:50.702Z [BLACKBORDER] Set mode to: default
2024-02-24T20:42:50.704Z [IMAGETOLED0] (ImageProcessor.cpp:161) Set LED mapping type to advanced
2024-02-24T20:42:50.704Z [IMAGETOLED0] (ImageProcessor.cpp:145) setSparseProcessing to 0
2024-02-24T20:42:50.707Z [RGB_TRANSFORM0] RGB transform classic_config: 1, saturationGain: 1.000000, luminanceGain: 1.000000, backlightThreshold: 1
2024-02-24T20:42:50.708Z [RGB_TRANSFORM0] setBacklightThreshold: 1
2024-02-24T20:42:50.709Z [ADJUST_RED0] Set correction to 255
2024-02-24T20:42:50.709Z [ADJUST_BLUE0] Set correction to 255
2024-02-24T20:42:50.709Z [ADJUST_GREEN0] Set correction to 255
2024-02-24T20:42:50.742Z [JSONSERVER] (JsonServer.cpp:24) Created instance
2024-02-24T20:42:50.750Z [JSONSERVER] Started on port 19444
2024-02-24T20:42:50.792Z [PIPEWIRE_SYSTEM:auto] (SystemWrapper.cpp:75) Closing grabber: PIPEWIRE_SYSTEM:auto
2024-02-24T20:42:50.794Z [X11_SYSTEM:auto] (SystemWrapper.cpp:75) Closing grabber: X11_SYSTEM:auto
2024-02-24T20:42:50.795Z [FRAMEBUF_SYSTEM:auto] (SystemWrapper.cpp:75) Closing grabber: FRAMEBUF_SYSTEM:auto
2024-02-24T20:42:50.720Z [HYPERHDR0] Led strip RGB order is: rgb
2024-02-24T20:42:50.721Z [DB] Database opened: /.hyperhdr/db/hyperhdr.db
2024-02-24T20:42:50.744Z [SMOOTHING0] High resolution clock is NOT STEADY!
2024-02-24T20:42:50.744Z [SMOOTHING0] Clearing queued colors before: enabling
2024-02-24T20:42:50.745Z [SMOOTHING0] Smoothing queue is cleared
2024-02-24T20:42:50.745Z [COMPONENTREG0] Smoothing: enabled
2024-02-24T20:42:50.745Z [SMOOTHING0] Creating config (0) => type: Alternative, dirMode: false, pause: false, settlingTime: 150ms, interval: 20ms (50Hz), antiFlickTres: 0, antiFlickStep: 2, antiFlickTime: 0
2024-02-24T20:42:50.746Z [SMOOTHING0] Clearing queued colors before: enabling. Smoothing configuration changed: restarting timer.
2024-02-24T20:42:50.746Z [SMOOTHING0] Smoothing queue is cleared
2024-02-24T20:42:50.746Z [SMOOTHING0] Selecting config (0) => type: Alternative, dirMode: false, pause: false, settlingTime: 150ms, interval: 20ms (50Hz), antiFlickTres: 0, antiFlickStep: 2, antiFlickTime: 0
2024-02-24T20:42:50.747Z [SMOOTHING0] Selecting config (0) => type: Alternative, dirMode: false, pause: false, settlingTime: 150ms, interval: 20ms (50Hz), antiFlickTres: 0, antiFlickStep: 2, antiFlickTime: 0
2024-02-24T20:42:50.748Z [LEDDEVICE_WS281X] Start LedDevice 'ws281x'.
2024-02-24T20:42:50.749Z [LEDDEVICE_WS281X] (LedDevice.cpp:171) deviceConfig: [{"colorOrder":"rgb","currentLedCount":183,"dma":5,"gpio":26,"invert":false,"leds":256,"pwmchannel":0,"refreshTime":0,"rgbw":false,"type":"ws281x","whiteAlgorithm":"subtract_minimum"}]
2024-02-24T20:42:50.749Z [LEDDEVICE_WS281X] (LedDevice.cpp:235) RefreshTime updated to 0ms
2024-02-24T20:42:50.750Z [LEDDEVICE_WS281X] (LedDeviceWS281x.cpp:59) ws281x strip type : 1050624
2024-02-24T20:42:50.752Z [LEDDEVICE_WS281X] (LedDevice.cpp:103) Enable device
2024-02-24T20:42:50.753Z [LEDDEVICE_WS281X] Device disabled, device 'ws281x' signals error: 'Failed to open. Error message: Hardware revision is not supported'
2024-02-24T20:42:50.754Z [MUXER0] Register new input 'System/VIDEOGRABBER' with priority 240 as inactive
2024-02-24T20:42:50.755Z [COMPONENTREG0] Video capture device: enabled
2024-02-24T20:42:50.796Z [V4L2:AUTO] Multithreading for V4L2 is enabled. Available thread's count 4
2024-02-24T20:42:50.797Z [V4L2:AUTO] Found capture device: /dev/video14
2024-02-24T20:42:50.798Z [V4L2:AUTO] Found capture device: /dev/video22
2024-02-24T20:42:50.800Z [V4L2:AUTO] Found capture device: /dev/video15
2024-02-24T20:42:50.801Z [V4L2:AUTO] Found capture device: /dev/video21
2024-02-24T20:42:50.803Z [V4L2:AUTO] (V4L2Grabber.cpp:192) Forcing auto discovery device
2024-02-24T20:42:50.803Z [V4L2:AUTO] Could not find any capture device 
2024-02-24T20:42:50.805Z [HYPERMANAGER] HyperHDR instance 'First LED instance' has been started
2024-02-24T20:42:50.807Z [DAEMON] Unload CEC
2024-02-24T20:42:50.760Z [FLATBUFSERVER] Started on port 19400
2024-02-24T20:42:50.762Z [FLATBUFSERVER] Started local domain socket server: 'hyperhdr-domain'
2024-02-24T20:42:50.762Z [PROTOSERVER] Started on port 19445
2024-02-24T20:42:50.766Z [SMOOTHING0] Using alternative smoothing procedure (0)
2024-02-24T20:42:50.807Z [WEBSERVER] Initialize Webserver
2024-02-24T20:42:50.807Z [WEBSERVER] Initialize Webserver
2024-02-24T20:42:50.972Z [WEBSERVER] Apply Webserver settings
2024-02-24T20:42:50.972Z [WEBSERVER] Set document root to: :/webconfig
2024-02-24T20:42:50.972Z [WEBSERVER] Apply Webserver settings
2024-02-24T20:42:50.973Z [WEBSERVER] Set document root to: :/webconfig
2024-02-24T20:42:50.975Z [WEBSERVER] Started on port 8090 name 'HyperHDR Webserver'
2024-02-24T20:42:51.019Z [WEBSERVER] Setup SSL certificate
2024-02-24T20:42:51.020Z [MUXER0] Set visible priority to 255
2024-02-24T20:42:51.020Z [WEBSERVER] Setup private SSL key
2024-02-24T20:42:51.020Z [HYPERHDR0] New priority[255], previous [255]
2024-02-24T20:42:51.021Z [HYPERHDR0] No source left -> switch LED-Device off
2024-02-24T20:42:51.021Z [WEBSERVER] Started on port 8092 name 'HyperHDR Webserver'
2024-02-24T20:42:51.021Z [SMOOTHING0] Clearing queued colors before: disabling
2024-02-24T20:42:51.021Z [SMOOTHING0] Smoothing queue is cleared
2024-02-24T20:42:51.021Z [LEDDEVICE_WS281X] (LedDevice.cpp:140) Disable device
2024-02-24T20:42:57.855Z [WEBSOCKET] (WebSocketClient.cpp:30) New connection from 2601:c6:c300:180:4956:20d4:dedf:d702
2024-02-24T20:42:57.856Z [WEBSOCKET] (JsonAPI.cpp:97) Client '2601:c6:c300:180:4956:20d4:dedf:d702' switch to HyperHDR instance 0
2024-02-24T20:42:58.224Z [DB] Database opened: /.hyperhdr/db/hyperhdr.db
2024-02-24T20:43:05.114Z [WEBSOCKET] (WebSocketClient.cpp:201) send close: 1000 
2024-02-24T20:43:06.222Z [WEBSOCKET] (WebSocketClient.cpp:30) New connection from 2601:c6:c300:180:4956:20d4:dedf:d702
2024-02-24T20:43:06.222Z [WEBSOCKET] (JsonAPI.cpp:97) Client '2601:c6:c300:180:4956:20d4:dedf:d702' switch to HyperHDR instance 0
2024-02-24T20:43:09.270Z [WEBSOCKET] (JsonAPI.cpp:1250) log streaming activated for client 2601:c6:c300:180:4956:20d4:dedf:d702

**Comment by awawa-dev** [comment]
1 https://github.com/awawa-dev/HyperHDR/wiki/FAQ
2 since RPi5 does not currently support it at all, it is not only an officially unsupported driver but a completely dead solution. Use HyperSerialPico instead (or HyperSerialESP32 or HyperSPI). I think that a several-year transition period in HyperHDR is more than enough to migrate, especially since I warned that it could end this way.
3 it's not HyperHDR driver but external https://github.com/jgarff/rpi_ws281x Try to use v20beta2 since it's using latest rpi_ws281x that was release ~10 months ago. It is possible that they added support for this board since for RPi3 it was enough to just change the number in the source code.
